src/GovUk.Education.ExploreEducationStatistics.Public.Data.Processor/local.settings.json

```diff
@@ -6,6 +6,7 @@
     "WEBSITE_TIME_ZONE": "GMT Standard Time"
   },
   "ConnectionStrings": {
-    "PublicDataDb": "Host=db;Username=postgres;Password=password;Database=public_data"
+    "PublicDataDb": "Host=db;Username=postgres;Password=password;Database=public_data",
+    "CoreStorage": "DefaultEndpointsProtocol=http;AccountName=devstoreaccount1;AccountKey=Eby8vdM02xNOcqFlqUwJPLlmEtlCDXJ1OUzFT50uSRZ6IFsuFq2UVErCz4I6tq/K1SZFPTOtr/KBHBeksoGMGw==;BlobEndpoint=http://data-storage:10000/devstoreaccount1;QueueEndpoint=http://data-storage:10001/devstoreaccount1;TableEndpoint=http://data-storage:10002/devstoreaccount1"
```

Comment:
Yeah, I was going to add a connection test into the Function App to prove the connectivity is there, but you're right in that I don't know if it'll actually be necessary or not at this point, as we might end up using file shares in a way that you don't need storage connection strings like this.  I'm tempted to leave it there for now, but just let Nick and Jack know that they're welcome to remove it should they want to.  Equally I'm happy to just remove it now.  No actually, I'm tempted to leave it and see how the Processor evolves!

Reply:
Ok, I should have mentioned another reason why I raised this is that I'm not sure "ConnectionStrings" is the right place 
to put non-database connection strings like storage account connection strings.

"ConnectionStrings" gets special treatment in the Configuration API and also has its own section within the "Application settings" tab of App Services and Function apps in Azure Portal. Entries there get displayed in a table showing each string alongside its type e.g. "SQLAzure".

See an example of where we define connection strings in the ARM template e.g. for the publisher app where `ConnectionStrings` has its own configuration section specifying the type of each string, rather than setting up connection strings as standard environment variables.

There is a "Custom" option for connection strings which suggests it _might_ be ok to add connection strings which don't relate to a database provider.

However, at some point in the past, if not still today, the way these "ConnectionStrings" environment variables get accessed is/was different in Azure Function app environments to how they are accessed in a local environment.

Reading the docs, it looks like there might be an automatic mapping between environment variables `"{Type}_{Key}"` and `"ConnectionStrings:{Key}"` now but we would need to test this in Azure.

Locally we can access connection strings like `Environment.GetEnvironmentVariable("ConnectionStrings:PublicDataDb")` as you would expect, but in an Azure environment, at least in the past they had to be accessed like `Environment.GetEnvironmentVariable("SQLAZURECONNSTR_PublicDataDb")`.

If nothing's changed today, this one would need to be accessed like `Environment.GetEnvironmentVariable($"CUSTOMCONNSTR_{name}"`, so this added complication is why none of our Function app projects use "ConnectionStrings" for storage account connection strings.

If this has changed then we could clean up or remove the code in [ConnectionUtils](https://github.com/dfe-analytical-services/explore-education-statistics/blob/7e86694472de4cdda3ae6bbea4b604cd614b1ba2/src/GovUk.Education.ExploreEducationStatistics.Common/Functions/ConnectionUtils.cs) which we wrote to make this seamless.

Reply:
So for now I have moved it out of ConnectionStrings and into a top-level config item, but retained it.  It's seconds to remove it but a bit more faff to add it in again, so I figured that it'd be worth retaining just for reference to allow, say, the Container App to adopt it as well.

As far as accessing the Connection Strings in the Function App, as you can see we're using the plain old `configuration.GetConnectionString("PublicDataDb")` as seen in Program.cs, without the need to access environment variables, and this is working in Azure as well as local.

Reply:
I'm not sure if that covers what you mentioned above - certainly it's a one-liner way of getting a connection string reference both locally and from Azure, and I've seen all the fun going off in the Util!